### PR TITLE
Multiple secrets: causes spec.volumes[].name duplication

### PR DESCRIPTION
**Pull request by jupierce**
Example:
could not run steps: step launch failed: failed to create or restart test pod: unable to create pod: Pod "launch" is invalid: spec.volumes[1].name: Duplicate value: "test-secret"

### CODE REVIEW

**Comment by alvaroaleman** [comment]
@jupierce mind adding a test for this?

File: pkg/steps/pod.go [hunk 1]
```diff
@@ -240,7 +240,7 @@ func (s *podStep) generatePodForStep(image string, containerResources coreapi.Re
 	pod.Spec.ServiceAccountName = s.config.ServiceAccountName
 	container := &pod.Spec.Containers[0]
 	for _, secret := range s.config.Secrets {
```
Comment (alvaroaleman) [inline pkg/steps/pod.go:242]: Is it safe to assume that the secret name here will be a valid kube secret name? Otherwise this might break a lot of things

Comment (jupierce) [reply pkg/steps/pod.go:242]: ack. Will add testcase. Naming works for our dataset, but I'll tighten the logic to work with any valid k8s.
/hold

**Comment by alvaroaleman** [comment]
Going to uncc, as this creates a lot of notifications, feel free to ping me once its ready for review
/uncc

**Comment by jupierce** [comment]
@alvaroaleman - cleaned up now.
- Removed reliance on secret name being a valid mount name.
- Out of an abundance of caution, the existing mount name and mount location are preserved for the first secret in a config. Only additional secrets will use the new convention.
- Tests have been added.

**Review by alvaroaleman — APPROVED** [review]

**Pull request merged**